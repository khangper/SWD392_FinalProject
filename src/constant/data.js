export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
