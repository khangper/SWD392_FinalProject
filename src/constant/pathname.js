export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
