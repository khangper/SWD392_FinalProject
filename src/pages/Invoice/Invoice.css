.Invoice_Container {
    background-color: lightgray;
    display: flex;
    padding: 0 350px 0 350px;
    flex-direction: column;
    height: 100%;
    height: 100vh;
    width: 100%;
}


.Invoice_Header {
    background: #333;
    float: left;
    width: 100%;
    height: 60px;
    display: flex !important;
    align-items: center !important;

}



.Invoice_Header_Item {
    float: left;
    width: 100%;
    text-align: center;
}

.Header_Logo {
    float: left;
    width: 135px;
}

.Header_Logo a img {
    width: 100%;

    padding: 25px 25px;
}


.Invoice_Header_Item p {
    font-size: 18px;
    font-family: 'Roboto', sans-serif;
    font-weight: 600;
    color: #fff;
    padding: 25px 30px;
    float: right;
}

/* p {
    margin: 0 0 1em;
    line-height: 1.4285em;
} */

.Invoice_Body {
    padding: 40px 0;
    background: #fff;
    float: left;
    width: 100%;
    height: 100%;
}

.Invoice_Date_Inf {
    float: right;
    width: 100%;
    margin-right: 60px;
}

.Invoice_Date_Inf ul {
    float: right;
}

.Invoice_Date_Inf ul li {
    display: block;
    margin-bottom: 5px;
}

.vdt-list {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    text-align: left;
    line-height: 24px;
}

.vdt-list span {
    color: #333;
    /* margin-right: 10px; */
}

.Invoice_Body_Content {
    float: left;
    width: 100%;
    margin-left: 40px;

}

.Inv_Content_Title {
    float: left;
    width: 100%;
    margin-top: 0;
    margin-bottom: 0;
    font-size: 16px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: left;
}

.Inv_Detail_Row {
    display: inline-flex;
    justify-content: space-between;
    width: 100%;

}

.Inv_Detail_Content-To,
.Inv_Detail_Content-Cursus {
    margin-top: -10px;
    width: 50%;
    display: flex;
    flex-direction: column;
}

.Inv_Detail_Content h4 {
    font-size: 16px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: left;
    margin-bottom: 15px;
}

.Invoice_Table {
    /* float: left; */
    width: 90%;
    margin-top: 50px;
    background: #fafafa;
    border-radius: 5px;
    margin-bottom: 0 !important;
    display: flex;
    flex-direction: column;
    margin: 0 50px 0 50px
}



.tb_head,
.user_dt_trans {
    display: flex;
    justify-content: space-between;
    padding: 15px 20px;
    border-bottom: 1px solid #efefef;
}

.tb_head {
    font-size: 14px;
    font-weight: 500;
    color: #333;
    font-family: 'Roboto', sans-serif;
    padding-bottom: 0;
}

.user_dt_trans {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    line-height: 24px;
    padding-top: 15px;
}

.tb_head p,
.user_dt_trans p {
    flex: 1;
    text-align: left;
    margin: 0;
    padding: 0 10px;
    /* Khoảng cách giữa các cột */
}

.tb_head p {
    font-weight: 500;
}

.user_dt_trans p {
    font-weight: 400;
}


.jsk1145 {
    display: flex;
    flex-direction: column;
    margin: 0px 0px 0 420px;
}

.jsk1145 .totalinv2 {
    font-size: 16px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #242424;
    margin-bottom: 10px;
}

.print_btn {
    float: right;
    background: #ed2a26;
    padding: 10px 30px 10px 30px;
    color: #fff;
    font-family: 'Roboto', sans-serif;
    font-weight: 500;
    font-size: 14px;
    text-align: center;
    text-transform: uppercase;
    border-radius: 3px;
}

.Invoice_Footer {
    width: 100%;
    margin-top: 30px;
    display: flex;
    justify-content: space-between;

}

.Left_Footer {
    width: 70%;
}

.Right_Footer {
    width: 30%;
    height: 10px;
    margin-right: 30px;
}



.Left_Footer p {
    font-size: 14px;
    color: #757575;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    text-align: left;
    margin-left: 30px;
}