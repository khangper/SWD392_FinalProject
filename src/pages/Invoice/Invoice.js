import React from "react";
import "./Invoice.css";
import { Link } from "react-router-dom";
import { PATH_NAME } from "../../constant/pathname";

const Invoice = () => {
  return (
    <div className="Invoice_Container">
      <div className="Invoice_Header">
        <div className="Invoice_Header_Item">
          <div className="Header_Logo">
            <Link to={PATH_NAME.HOME}>
              <img
                src="https://gambolthemes.net/html-items/cursus-new-demo/images/ct_logo.svg"
                alt=""
              />
            </Link>
          </div>
          <p>Invoice</p>
        </div>
      </div>
      <div className="Invoice_Body">
        <div className="Invoice_Date_Inf">
          <ul>
            <li>
              <div className="vdt-list">
                <span>Date:</span>
                10 April 2020
              </div>
            </li>
            <li>
              <div className="vdt-list">
                <span>Invoice No:</span>
                IVIP12023598
              </div>
            </li>
            <li>
              <div className="vdt-list">
                <span>Order ID:</span>
                1258963487
              </div>
            </li>
          </ul>
        </div>
        <div className="Invoice_Body_Content">
          <div className="Inv_Content_Title">
            <h2>Invoice</h2>
          </div>
          <div className="Inv_Detail_Row">
            <div className="Inv_Detail_Content-To">
              <h4>To</h4>
              <div className="vdt-list">Rock William</div>

              <div className="vdt-list">133, Dracut</div>

              <div className="vdt-list">Massachusetts</div>

              <div className="vdt-list">01826</div>

              <div className="vdt-list">United States</div>
            </div>
            <div className="Inv_Detail_Content-Cursus">
              <h4>Cursus</h4>
              <div className="vdt-list">Cursus LTD</div>
              <div className="vdt-list">#1234, Shahid karnail Singh Nagar,</div>
              <div className="vdt-list">Near MBD Mall,</div>
              <div className="vdt-list">141001</div>
              <div className="vdt-list">Ludhiana </div>
              <div className="vdt-list">Punjab </div>
              <div className="vdt-list">India </div>
            </div>
          </div>
        </div>
        <div className="Invoice_Table">
          <div className="tb_head">
            <p>Item</p>
            <p>Price</p>
            <p>Vat(20%)</p>
            <p>Total Amount</p>
          </div>
          <div className="user_dt_trans">
            <p>Item Tittle</p>
            <p>$79.00</p>
            <p>$10.00</p>
            <p>$89.00</p>
          </div>
          <div className="user_dt_trans jsk1145">
            <div className="totalinv2">Invoice Total : USD $220.00</div>
            <p>Paid via Paypal</p>
          </div>
        </div>
        <div className="Invoice_Footer">
          <div className="Left_Footer">
            <p>Thanks for buying</p>
          </div>
          <div className="Right_Footer">
            <a className="print_btn" href="#">
              Print
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Invoice;
