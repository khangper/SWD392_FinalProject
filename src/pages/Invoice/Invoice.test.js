import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import Invoice from "./Invoice";
import { screen, waitFor, fireEvent } from "@testing-library/react";

describe("Invoice", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the Invoice component", async () => {
        const { container } = renderWithProviders(<Invoice />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });

});
