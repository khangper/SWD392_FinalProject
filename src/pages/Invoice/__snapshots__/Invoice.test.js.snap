// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice renders the Invoice component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`ReportHistory renders the Invoice component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
